doc/src/manual/performance-tips.md

```diff
@@ -358,6 +358,34 @@ julia> !isconcretetype(Array), !isabstracttype(Array), isstructtype(Array), !isc
 ```
 In this case, it would be better to avoid declaring `MyType` with a field `a::Array` and instead declare the field as `a::Array{T,N}` or as `a::A`, where `{T,N}` or `A` are parameters of `MyType`.
 
+The previous advice is especially useful when the fields of a struct are meant to be functions, or more generally callable objects.
+It is very tempting to define a struct as follows:
+
+```julia
+struct MyCallableWrapper
+    f::Function
+end
+```
+
+But since `Function` is an abstract type, every call to `wrapper.f` will be very inefficient.
```

Comment:
```suggestion
But since `Function` is an abstract type, every call to `wrapper.f` will require dynamic dispatch, due to
the type instability of accessing the field `f`.
```

I think it's good to explicitly mention the type instability here, since that's what we'd generally want people to look for anyway.